### PR TITLE
Remove two unused methods in GameData.java

### PR DESCRIPTION
**Pull request by DanVanAtta**

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/data/GameData.java [hunk 1]
```diff
@@ -293,14 +293,6 @@ public void removeDataChangeListener(final GameDataChangeListener listener) {
...
   }
 
-  public void addGameMapListener(final GameMapListener listener) {
-    gameMapListeners.add(listener);
-  }
-
-  public void removeGameMapListener(final GameMapListener listener) {
-    gameMapListeners.remove(listener);
-  }
-
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/data/GameData.java]: Does this mean there are no listeners at all? 
In this case all operations on gameMapListeners could be removed.
We need to keep the variable for backwards compatibility though.

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/data/GameData.java]: Following up here, variable access is not encapsulated by just these accessor methods. I tracked down a few more usages to:
```
  void notifyMapDataChanged() {
    gameMapListeners.forEach(GameMapListener::gameMapDataChanged);
  }

  public void clearAllListeners() {
    dataChangeListeners.clear();
    territoryListeners.clear();
    gameMapListeners.clear();
    if (resourceLoader != null) {
      resourceLoader.close();
      resourceLoader = null;
    }
  }
```

I think that probably hints why the methods are unused.

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/data/GameData.java]: Actually.. wait.. I think you're  right here  @RoiEXLab; taking a second look at this. :thinking:

**Comment by RoiEXLab** [comment]
Feel free to merge once the build completes

**Pull request merged**